test/models/key_pair_test.rb

```diff
@@ -0,0 +1,12 @@
+require 'test_helper'
+
+class KeyPairTest < ActiveSupport::TestCase
+  should belong_to(:compute_resource)
+
+  test "ensure validations" do
+    key = KeyPair.new
+    refute key.valid?
+    assert_equal([:name, :secret, :compute_resource_id], key.errors.messages.keys)
+    assert_equal(["can't be blank"], key.errors.messages.values.first.compact)
+  end
+end
```

Comment:
could you please add tests to the ComputeResourceKeyPairs::Base class? especially around active and used elsewhere?